VAMobile/src/api/claimsAndAppeals/downloadEFolderDocument.tsx

```diff
@@ -0,0 +1,42 @@
+import FileViewer from 'react-native-file-viewer'
+
+import { useQuery } from '@tanstack/react-query'
+
+import store from 'store'
+import { DEMO_MODE_LETTER_ENDPOINT } from 'store/api/demo/letters'
+import getEnv from 'utils/env'
+import { downloadDemoFile, downloadFile } from 'utils/filesystem'
+import { registerReviewEvent } from 'utils/inAppReviews'
+
+import { claimsAndAppealsKeys } from './queryKeys'
+
+const { API_ROOT } = getEnv()
+
+/**
+ * Fetch user E Folder Document
+ */
+const downloadEFolderDocument = async (id: string, fileName: string): Promise<boolean | undefined> => {
+  const eFolderDocumentAPI = `${API_ROOT}/v0/efolder/documents/${id}/download?file_name=${fileName}}`
+
+  const filePath = store.getState().demo.demoMode
+    ? await downloadDemoFile(DEMO_MODE_LETTER_ENDPOINT, fileName)
+    : await downloadFile('POST', eFolderDocumentAPI, fileName, undefined, 1)
+  if (filePath) {
+    await FileViewer.open(filePath, { onDismiss: () => registerReviewEvent() })
+    return true
+  }
+}
+
+/**
+ * Returns a query for a user E Folder Document
+ */
+export const useDownloadEFolderDocument = (id: string, fileName: string) => {
+  return useQuery({
+    enabled: false,
```

Comment:
Is this query disabled because it's not in use yet? If that's the case, I don't think we can QA it

Reply:
this first ticket is implementing the api and not QA able per Binny. The following ticket is implementation and that one will QA this one technically. This is disabled because I manually call fetch on state updates as you will see in the next pr so that fetching doesn't happen on loading of the screen, but after a user clicks on a file to download.